grails-app/services/io/xh/toolbox/MonitorDefinitionService.groovy

```diff
@@ -5,25 +5,103 @@ import io.xh.hoist.monitor.MonitorResult
 
 import static io.xh.hoist.monitor.MonitorStatus.OK
 import static io.xh.hoist.monitor.MonitorStatus.FAIL
+import static io.xh.hoist.monitor.MonitorStatus.UNKNOWN
+import static io.xh.hoist.monitor.MonitorStatus.WARN
 import static io.xh.hoist.util.DateTimeUtils.MINUTES
 import static java.lang.System.currentTimeMillis
 
 class MonitorDefinitionService extends BaseService {
 
     def newsService
+    def fileManagerService
+    def recallsService
 
+    /**
+     * Check the count of news stories loaded by NewsService
+     * @param result
+     * @return
+     */
     def newsStoryCount(MonitorResult result) {
         result.metric = newsService.itemCount
     }
 
+    /**
+     * Check when the last update to the news was fetched.
+     * If no news stories have been fetched at all, we consider that a failure.
+     * @param result
+     * @return
+     */
     def lastUpdateAgeMins(MonitorResult result) {
-        def diffMs = currentTimeMillis() - newsService.lastTimestamp.time,
-            diffHours = Math.floor(diffMs / MINUTES)
-        result.metric = diffHours
+        if (newsService.lastTimestamp) {
+            def diffMs = currentTimeMillis() - newsService.lastTimestamp.time,
+                diffHours = Math.floor(diffMs / MINUTES)
+            result.metric = diffHours
+        } else {
+            result.metric = -1;
+            result.status = FAIL
+            result.message = 'Have not yet loaded any stories'
+        }
     }
 
+    /**
+     * Check whether or not the NewsService has loaded stories from all its sources.
+     * @param result
+     * @return
+     */
     def loadedSourcesCount(MonitorResult result) {
         result.metric = newsService.loadedSourcesCount
         result.status = newsService.allSourcesLoaded ? OK : FAIL
     }
+
+    /**
+     * Check the storage space used by uploaded files in the FileManager app, in megabytes
+     * @param result
+     * @return
+     */
+    def storageSpaceUsed(MonitorResult result) {
+        //sum up the sizes of all uploaded files..
+        def bytes = fileManagerService.list()
+                .collect {it.length()}
+                .sum() ?: 0
+        //and convert to megabytes rounded to two decimal places
+        def MiB = (bytes / (1024 * 1024)).setScale(2, BigDecimal.ROUND_HALF_UP)
+        result.metric = MiB
+    }
+
+    /**
```

Comment:
I am not a fan of this monitor.  Think it raises more questions then we want to.